src/backend/FrozenLakeEnv.js

```diff
@@ -0,0 +1,118 @@
+/* Based on the OpenAI Gym implementation
+ * https://github.com/openai/gym/blob/master/gym/envs/toy_text/frozen_lake.py
+ */
+
+export class FrozenLakeEnv {
+	/**
+	 * Env Class
+	 */
+
+
+
+	constructor(desc=null, map_name="4x4", is_slippery=true) {
+
+	  this.MAPS = {
+        "4x4": [
+            "SFFF",
+            "FHFH",
+            "FFFH",
+            "HFFG"
+        ],
+        "8x8": [
+            "SFFFFFFF",
+            "FFFFFFFF",
+            "FFFHFFFF",
+            "FFFFFHFF",
+            "FFFHFFFF",
+            "FHHFFFHF",
+            "FHFFHFHF",
+            "FFFHFFFG"
+        ],
+    };
+
+    if (desc === null && map_name === null) {
+      this.desc = this.MAPS["4x4"];
+    } else if (desc===null) {
+      this.desc = this.MAPS[map_name];
+    }
+
+    this.nrow = this.desc.length;
+    this.ncol = this.desc[0].length;
+    this.reward_range = (0, 1);
+
+    this.nA = 4;
+    this.nS = this.nrow * this.ncol;
+
+    this.reset();
+  }
+
+	// Returns a tensor describing the state space of the env
+	// Will be used to determine the input dimensionality of the model
+	getStateSpace() {
+    return 2;
+  }
+
+	// Returns a tesor describing the action space of the env
+	// Will be used to determine the output dimensionality of the model
+	getActionSpace() {
+    return 4;
```

Comment:
Same as above, should return `this.nA`?

Reply:
Maybe return a vector with the possible actions? I know we discussed this yesterday. If I recall correctly, you only wanted the number of actions. In that case, `this.nA` would do the work.

Reply:
Yeah, same as above I really just want something to stick into the models. Integer number of actions would be easiest